### ISSUE TITLE
Can we use EBM classifier to explain xgboost classifier?

### ISSUE THREAD

**Issue by Saurabhdhoble-8451** [interpretml/interpret#466]
Hi Team, 
               I was trying to understand if we could leverage model explanation capability of EBM's and apply it to explain black box xgboost model.Tried below code.

Step1)Trained/fitted an xgboost model
Step2)Extracted a lower level xgboost model using xgb_fit.booster
Step3)Passed this model to EBM classifier.
Step4)Did a fit on explainer but now it has model information of fitted xgboost model.

 explainer = ExplainableBoostingClassifier(xgb_fit.booster,n_jobs=16,inner_bags=25,outer_bags=50)
 ##Fit explainer
 ebm = explainer.fit(X_train, y_train)

 #explain the model
 explanation = explainer.explain_global(X_test)

My understanding:
The explainer sees it has been passed a raw booster object, not a sklearn model.
It will use this booster to generate predictions on the training data by calling xgb_fit.booster.predict(X_train)
These predictions will serve as the target for training the EBM explainer model.
The explainer will fit an interpretable EBM model to try to mimic the predictions from the complex xgb booster.
So in summary, the raw booster object is only used to obtain predictions from the fitted XGB model. The explainer itself operates by fitting a separate interpretable model on top of these predictions. The booster does not get directly incorporated into the EBM explainer.

Please let me know if my understanding is correct and this approach could be used.

**Comment by paulbkoch** [comment]
Hi @Saurabhdhoble-8451 -- The ExplainableBoostingClassifier init function does not accept another model.  If you want to mimic an xgboost model, you can do this by generating a "y_predict" first from the xgboost model, then train the EBM by passing in y_predict instead of y_train.

**Comment by Saurabhdhoble-8451** [comment]
@paulbkoch ...thanks for your response paul.This helps.

**Comment by Saurabhdhoble-8451** [comment]
@paulbkoch's response helped.
